src/Persistence/MassTransit.RedisIntegration/RedisSagaRepository.cs

```diff
@@ -126,14 +126,30 @@ public void Probe(ProbeContext context)
 
         async Task UpdateRedisSaga(TSaga instance)
         {
-            ITypedDatabase<TSaga> sagas = _redisDbFactory().As<TSaga>();
+            RedisKey key = $"saga_lock_{instance.CorrelationId}";
+            RedisValue token = Environment.MachineName + Guid.NewGuid();
+            var db = _redisDbFactory();
+            ITypedDatabase<TSaga> sagas = db.As<TSaga>();
+            if(db.LockTake(key, token, TimeSpan.FromMinutes(1)))
+            {
+                try
```

Comment:
Not exactly. There was no `try` and therefore no `catch`, so if it fails - it would be caught by the retry filter and retried. If the exception is silently ignored - in case of any Redis or network failure there will be a data loss.

Reply:
How so? If there is no catch the exception isn\`t caught.
Try-finally block only provides for lock release even in case of an exception.

Reply:
A little sample that proof the try-finally block behavior

```
try
{
    try
    {
        throw new Exception();
    }
    finally
    {
        Console.WriteLine("Finally");
    }
}
catch (Exception ex)
{
    Console.WriteLine("Catch");
}
```

Reply:
Yeah, you are right. I thought there is an empty catch.